astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
All I was saying is:
Consider a FITS file in which hdu's have a checksum. In such a case if I open it with
- `checksum=True` it will proceed with verification and correction if possible, 
-  if `checksum='remove'` it will remove the keywords when writing the changes (via `writeto`)
-  if `checksum=False` I understand that it should not proceed with verification, but does that mean `_checksum` property should remain **unset** (even if keywords were present).